IntegrationTests/common/script/Makefile.mk

```diff
@@ -77,7 +77,8 @@ $(DEPS)/functions/%.d: $(EMDATA)/MemPrints
 	 TOP_FUNC=`echo $@ | sed 's,.*\/\([^/]*\)\.d,\1,g'`; \
 	 TOP_FILE=`grep -l $${TOP_FUNC} $(TOP_FUNCS)/*.cc | sed 's,.*\/\([^/]*\)\.cc,\1,g'`; \
 	 echo "$${TOP_FUNC}: $(DEPS)/files/$${TOP_FILE}.d $(EMDATA)/MemPrints" > $@; \
-	 echo "	vivado_hls -f $(COMPILE_HLS) $(TOP_FUNCS)/$${TOP_FILE}.cc $${TOP_FUNC}" >> $@; \
+	 echo "	@rm -rfv $${TOP_FUNC}; \\" >> $@; \
+	 echo "	 vivado_hls -f $(COMPILE_HLS) $(TOP_FUNCS)/$${TOP_FILE}.cc $${TOP_FUNC}" >> $@; \
```

Comment:
Not to nit-pick, but shouldn't this be a rule in the makefile for the appropriate file endings rather than generating a rule  for each dependency explicitly?  something like 
```
#
# The rule for building the object file from each C source file.
#
${TOP_FUNCS}/${SUFFIX}/%.o: %.cc
	vivado_hls -f $(COMPILE_HLS) $(TOP_FUNCS)/$${TOP_FILE}.cc $${TOP_FUNC}

```

Reply:
The issue is determining the file that a given top function is contained in. Either this mapping between function and file has to be hard-coded in the Makefile itself, or it has to be determined dynamically, which is what is done now with `grep` (N.B., TOP_FUNC and TOP_FILE are shell variables of the recipe, not Makefile variables).

However, I did just realize that the two pattern rules in Makefile.mk can be collapsed into just one, which I think is easier to understand.